src/main/java/upb/dice/twitter/MaxIDChecker.java

```diff
@@ -19,51 +23,59 @@
-                map.put(splitString[0], splitString[1]);
-            }
-            if (map.containsKey(key)) {
-                String id_string = map.get(key);
-                return new BigDecimal(id_string).intValue();
-            }
-            else {
-                return 1L;
-            }
+        File file = new File("Tweets Search Details" + File.separator + "Keyword_MaxID.txt");
+        file.getParentFile().mkdirs();
+        file.createNewFile();
+        BufferedReader br = new BufferedReader(new FileReader("Tweets Search Details" + File.separator + "Keyword_MaxID.txt"));
+        String line;
+        while ((line = br.readLine()) != null) {
+            String[] splitString = line.split(",");
+            map.put(splitString[0], splitString[1]);
         }
+        if (map.containsKey(key)) {
+            String id_string = map.get(key);
+            result = new BigDecimal(id_string).toBigInteger().longValue();
+        } else {
+            result = 1L;
+        }
+        return result;
     }
-
     /**
      * This method checks for the maxID for the current geoLocation. Returns 1 if geoLocation is new
+     *
      * @param geoLocation for which the maxID has to be checked
      * @return the maxID for the respective geoLocation
      */
     public long locationParse(GeoLocation geoLocation) throws IOException {
+        long result = 1L;
         Map<List<String>, String> pairStringMap = new HashMap<>();
-        try (BufferedReader br = new BufferedReader(new FileReader("Latitude_Longitude_MaxID1.txt"))) {
-            String line;
-            while ((line = br.readLine()) != null) {
-                String[] splitString = line.split(",");
-                String latitude = splitString[0];
-                String longitude = splitString[1];
-                String locMaxId = splitString[3];
-                pairStringMap.put(Arrays.asList(latitude, longitude), locMaxId);
-            }
-            String lon_str = String.valueOf(geoLocation.getLatitude());
-            String lat_str = String.valueOf(geoLocation.getLatitude());
-            if (pairStringMap.containsKey(Arrays.asList(lon_str, lat_str))) {
-                String id_string = pairStringMap.get(Arrays.asList(lon_str, lat_str));
-                return (new BigDecimal(id_string).intValue());
-            }
-            return 1L;
+        File file = new File("Tweets Search Details" + File.separator + "Latitude_Longitude_MaxID.txt");
+        file.getParentFile().mkdir();
+        file.createNewFile();
+        String line;
+        BufferedReader br = new BufferedReader(new FileReader(file));
+        while ((line = br.readLine()) != null) {
+            String[] splitString = line.split(",");
+            String latitude = splitString[0];
+            String longitude = splitString[1];
+            String locMaxId = splitString[3];
+            pairStringMap.put(Arrays.asList(latitude, longitude), locMaxId);
```

Comment:
Did you checked whether it really works? I am not sure whether the `ArrayList` you are creating here can be used as a key for a Map.

Reply:
Why do you even have to create the map? Since you already know what you are searching for at this point in time, you could simply search directly here and save time (and memory).

Reply:
> Why do you even have to create the map? Since you already know what you are searching for at this point in time, you could simply search directly here and save time (and memory).

The files "Keyword_MaxID.txt" and "Location_MaxID.txt" contains multiple maxIDs for the same keyword and location respectively. Creating a map helps to have the unique and latest maxID.

Reply:
> Did you checked whether it really works? I am not sure whether the `ArrayList` you are creating here can be used as a key for a Map.

Yes, it is working